### PR TITLE
[Core] Wrong/strange reaction computation with elimination builder and solver fixed

### LINKED ISSUE #3442: Wrong/strange reaction computation with elimination builder and solver
I see some strange reaction calculation when using Elimination builder and solver. 
Below is the case attached. (with elimination builder and solver)
Also for initial impression I am adding two screen shots : first one with block builder and solver , second one with elimination builder and solver

block_builder_solver_reaction_plot

elem_builder_solver_reaction_plot

[34_test_elem_b_and_s.gid.zip](https://github.com/KratosMultiphysics/Kratos/files/2604169/34_test_elem_b_and_s.gid.zip)

### PR DESCRIPTION
**Pull request by loumalouomega**
Fixes #3442

### CODE REVIEW

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h [hunk 1]
```diff
@@ -886,18 +886,15 @@ class ResidualBasedEliminationBuilderAndSolver
...
-        int systemsize = BaseType::mDofSet.size() - TSparseSpace::Size(*BaseType::mpReactionsVector);
-
-        typename DofsArrayType::ptr_iterator it2;
-
-        //updating variables
-        TSystemVectorType& ReactionsVector = *BaseType::mpReactionsVector;
-        for (it2 = BaseType::mDofSet.ptr_begin(); it2 != BaseType::mDofSet.ptr_end(); ++it2)
-        {
+        // Updating variables
+        std::size_t i;
```
Comment (adityaghantasala) [inline kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h:890]: IndexType ?

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h [hunk 2]
```diff
@@ -864,9 +864,9 @@ class ResidualBasedEliminationBuilderAndSolver
         //if needed resize the vector for the calculation of reactions
         if (BaseType::mCalculateReactionsFlag == true)
         {
-            unsigned int ReactionsVectorSize = BaseType::mDofSet.size();
-            if (BaseType::mpReactionsVector->size() != ReactionsVectorSize)
-                BaseType::mpReactionsVector->resize(ReactionsVectorSize, false);
+            const std::size_t reactions_vector_size = BaseType::mDofSet.size() - BaseType::mEquationSystemSize;
```
Comment (adityaghantasala) [inline kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h:867]: const IndexType ?

**Review by adityaghantasala — APPROVED** [review]
Works as it should, just two minor comments !

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h (hunk 2 above)
Comment (loumalouomega) [reply kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h:867]: In this version of the B&S the typedef is not defined

File: kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h (hunk 1 above)
Comment (loumalouomega) [reply kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver.h:890]: Idem

**Pull request merged**